Source/WebCore/workers/service/ServiceWorkerRegistration.cpp

```diff
@@ -319,6 +316,12 @@ void ServiceWorkerRegistration::showNotification(ScriptExecutionContext& context
             auto& jsPromise = *JSC::jsCast<JSC::JSPromise*>(promise->promise());
             pushEvent->waitUntil(DOMPromise::create(globalObject, jsPromise));
         }
+
+        if (!serviceWorkerGlobalScope->hasPendingSilentPushEvent() && serviceWorkerGlobalScope->didFirePushEventRecently())
+            serviceWorkerGlobalScope->addConsoleMessage(MessageSource::Storage, MessageLevel::Warning, "showNotification was called outside of push event lifetime"_s, 0);
```

Comment:
Should this message also mention that you may want to pass the promised returned by `showNotification` to the push event's `waitUntil` to appropriately extend the lifetime of the push event?

Reply:
OK, I'll add a mention to PushEvent.waitUntil, trying to stick with a short version.